src/routes/safe/components/Transactions/TxList/BatchExecute.tsx

```diff
@@ -0,0 +1,196 @@
+    setModalOpen((prevOpen) => !prevOpen)
+  }
+
+  const handleOpenModal = () => {
+    toggleModal()
+
+    const handleGetBatchExecuteData = async () => {
+      const batchExecuteData = await getBatchExecuteData(
+        dispatch,
+        batchableTransactions,
+        safeInstance,
+        safeAddress,
+        account,
+      )
+      setButtonStatus(ButtonStatus.READY)
+      setMultiSendCallData(batchExecuteData)
+    }
+    handleGetBatchExecuteData()
+  }
+
+  const handleBatchExecute = async () => {
+    createMultiSendTransaction({
+      transactions: batchableTransactions,
+      multiSendCallData,
+      dispatch,
+      account,
+      safeAddress,
+    })
+
+    toggleModal()
+  }
+
+  return (
+    <>
+      <StyledButton
+        color="primary"
+        variant="contained"
+        onClick={handleOpenModal}
+        disabled={batchableTransactions.length <= 1 || hasPendingTx}
+        onMouseEnter={handleOnMouseEnter}
+        onMouseLeave={handleOnMouseLeave}
+      >
+        Batch-Execute {batchableTransactions.length > 1 ? `(${batchableTransactions.length})` : ''}
+      </StyledButton>
+      <Modal description="Batch-Execute" handleClose={toggleModal} open={isModalOpen} title="Batch-Execute">
+        <ModalHeader onClose={toggleModal} title="Batch-execute transactions" />
+        <Hairline />
+        <ModalContent>
+          <Row margin="md">
+            <Paragraph noMargin>This action will execute {batchableTransactions.length} Transactions.</Paragraph>
+          </Row>
+          <Row margin="xs">
+            <Paragraph color="disabled" noMargin size="lg">
+              {getInteractionTitle('0')}
+            </Paragraph>
+          </Row>
+          <Row margin="md">
+            <PrefixedEthHashInfo
+              hash={multiSendContractAddress}
+              showAvatar
+              textSize="lg"
+              showCopyBtn
+              explorerUrl={getExplorerInfo(multiSendContractAddress)}
+            />
+          </Row>
+          <Paragraph size="md" align="center" color="disabled" noMargin>
+            Be aware that if any of the transactions fail, all of them will revert.
+          </Paragraph>
+        </ModalContent>
+        <Modal.Footer withoutBorder>
+          <Modal.Footer.Buttons
+            cancelButtonProps={{ onClick: toggleModal, text: 'Cancel' }}
+            confirmButtonProps={{
+              status: buttonStatus,
+              onClick: handleBatchExecute,
+              disabled: batchableTransactions.length <= 1 || buttonStatus === ButtonStatus.LOADING,
+              text: buttonStatus === ButtonStatus.LOADING ? 'Loading' : 'Submit',
+            }}
+          />
+        </Modal.Footer>
+      </Modal>
+    </>
+  )
+}
+
+const StyledButton = styled(Button)`
+  align-self: flex-end;
+  margin-right: ${sm};
+  margin-top: ${md};
+`
+
+const ModalContent = styled.div`
+  padding: ${lg} ${lg} 0;
+`
+
+export const MemoizedBatchExecute = React.memo(BatchExecute)
```

Comment:
Add a comment above the component to express that this is memoized. If we ever need to add `props` here, this would cause issues.